src/economy/portdock.cc

```diff
@@ -502,6 +382,32 @@ uint32_t PortDock::count_waiting(const PortDock* dest) const {
 	return waiting_.size();
 }
 
+uint32_t PortDock::calc_max_priority(const EditorGameBase& egbase, const PortDock& dest) const {
+	uint32_t p = 0;
+	for (const ShippingItem& si : waiting_) {
+		if (si.destination_dock_.serial() == dest.serial()) {
+			WareInstance* ware = nullptr;
+			Worker* worker = nullptr;
+			si.get(egbase, &ware, &worker);
+			++p;
+			if (ware) {
+				assert(!worker);
+				if (ware->get_transfer() && ware->get_transfer()->get_request()) {
+					// I don't know when this shouldn't be true,
+					// but the regression tests assure me that it's possible…
+					p += ware->get_transfer()->get_request()->get_transfer_priority();
+				}
+			} else {
+				assert(worker);
+				if (worker->get_transfer() && worker->get_transfer()->get_request()) {
+					p += worker->get_transfer()->get_request()->get_transfer_priority();
+				}
```

Comment:
As worker and ware are handled the same is ther no commen superclass to handle this?

Reply:
Every Transfer has a priority assigned to it. Higher priority means that it should be transported as soon as possible. I am taking this into account both here and in the comment above so that e.g. construction materials are prioritised over wares that will just be stored in a warehouse.
A Transfer will have different priorities depending on whether it handles a ware or worker, so no case distinction needed here.